### PR TITLE
Rework UI

### LINKED ISSUE #193: `Bug`: List of minor UI-Bugs
I'll just start a discussion on minor UI-Bugs noticed while correcting. Feel free to add the ones you found down below :)

1) Line numbers are not visible by default (when using the build from github) ✅ 
2) Test-Results is not scrollable by mouse-wheel

### PR DESCRIPTION
**Pull request by dfuchss**
* I've reworked the Test UI . Scrolling and so on might work better now.
* Also reworked the Assessment Tab / Grading Tab Improvements are welcome :)

This will finally close #193 

Files: https://github.com/kit-sdq/programming-lecture-eclipse-artemis/actions/runs/2307752727

### CODE REVIEW

**Comment by dfuchss** [comment]
@Shirkanesi are the parts that shall be scrollable now scrollable :) #193

**Comment by dfuchss** [comment]
The UI parts are compatible with https://www.eclipse.org/windowbuilder/ so if someone wants to change anything here, please us the window builder.

**Comment by Shirkanesi** [comment]
> @Shirkanesi are the parts that shall be scrollable now scrollable :) #193

Scrolling works for me :)

But: I'm sad you removed the grading-tab. For my workflow, I used the Outline-Tab quite often for larger classes and I also like its location in the bottom-right corner. However with the redesign having the Outline open at a reasonable size I've got to scroll to see all the penalty-buttons.
So not realy a big fan of that. 
Not a dealbreaker but should be kept in mind. Especially if peoply like a wide code-editor they might only used 2 columns for the buttons what would make the scrolling even more annoying.

Also not quite shure what happened but the exercise-combo does not list all exercises (I can only see 3 but 7 in the old version :().

**Comment by dfuchss** [comment]
@Shirkanesi the grading tab should be visible again :)

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/preferences/AdvancedPreferences.java [hunk 1]
```diff
@@ -40,8 +41,14 @@ public void createFieldEditors() {
 		var overrideDefaultPreferences = new BooleanFieldEditor(PreferenceConstants.OVERRIDE_DEFAULT_PREFERENCES, "Tweak Eclipse Preferences on startup",
 				parent);
 
-		this.addField(absoluteConfigPath);
+		var columnsForGradingButtons = new IntegerFieldEditor(PreferenceConstants.GRADING_BUTTONS_IN_COLUMN, "Amount of Grading Buttons in one column", parent);
```
Comment (Shirkanesi) [inline bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/preferences/AdvancedPreferences.java:44]: Wouldn't this technically be "in one row"? :D

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/utilities/UIUtilities.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/* Licensed under EPL-2.0 2022. */
+package edu.kit.kastel.eclipse.common.view.utilities;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.ScrolledComposite;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.TabFolder;
+import org.eclipse.swt.widgets.TabItem;
+
+public class UIUtilities {
```
Comment (Shirkanesi) [inline bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/utilities/UIUtilities.java:10]: `final` :D

**Review by Shirkanesi — CHANGES_REQUESTED** [review]

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/preferences/AdvancedPreferences.java (hunk 1 above)
Comment (dfuchss) [reply bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/preferences/AdvancedPreferences.java:44]:
```suggestion
		var columnsForGradingButtons = new IntegerFieldEditor(PreferenceConstants.GRADING_BUTTONS_IN_COLUMN, "Amount of Grading Buttons in one row", parent);
```

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/utilities/UIUtilities.java (hunk 2 above)
Comment (dfuchss) [reply bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/utilities/UIUtilities.java:10]:
```suggestion
public final class UIUtilities {
```

**Review by Shirkanesi — APPROVED** [review]

**Pull request merged**